crates/holochain/tests/tests/conductor/mod.rs

```diff
@@ -0,0 +1,341 @@
+            agent_key: None,
+            source: AppBundleSource::Bundle(bundle.clone()),
+            installed_app_id: Some("app_1".into()),
+            network_seed: Some(network_seed_override.into()),
+            roles_settings: Some(HashMap::from([role_settings])),
+            ignore_genesis_failure: false,
+            allow_throwaway_random_agent_key: true,
+        })
+        .await
+        .unwrap();
+
+    // - Check that the dna hash differs between the app installed with and the one installed without
+    //   custom modifiers
+
+    let app_info_0 = conductor
+        .get_app_info(&"app_0".to_string())
+        .await
+        .unwrap()
+        .unwrap();
+
+    let dna_hash_0 = match app_info_0
+        .cell_info
+        .into_iter()
+        .find(|(role_name, _)| role_name == &role_name_1)
+        .unwrap()
+        .1[0]
+        .clone()
+    {
+        CellInfo::Provisioned(c) => c.cell_id.dna_hash().clone(),
+        _ => panic!("wrong cell type."),
+    };
+
+    let app_info_1 = conductor
+        .get_app_info(&"app_1".to_string())
+        .await
+        .unwrap()
+        .unwrap();
+
+    let dna_hash_1 = match app_info_1
+        .cell_info
+        .into_iter()
+        .find(|(role_name, _)| role_name == &role_name_1)
+        .unwrap()
+        .1[0]
+        .clone()
+    {
+        CellInfo::Provisioned(c) => c.cell_id.dna_hash().clone(),
+        _ => panic!("wrong cell type."),
+    };
+
+    assert_ne!(dna_hash_0, dna_hash_1);
+
+    let manifest = app_info_1.manifest;
+
+    // - Check that the modifers have been set correctly and only for the specified role
+    let installed_app_role_1 = manifest
+        .app_roles()
+        .into_iter()
+        .find(|r| r.name == role_name_1)
+        .unwrap();
+
+    let installed_app_role_2 = manifest
+        .app_roles()
+        .into_iter()
+        .find(|r| r.name == role_name_2)
+        .unwrap();
+
+    assert_eq!(
+        installed_app_role_1.dna.modifiers.network_seed,
+        Some(custom_network_seed)
+    );
+    assert_eq!(
+        installed_app_role_1.dna.modifiers.properties,
+        Some(custom_properties)
+    );
+    assert_eq!(
+        installed_app_role_1.dna.modifiers.origin_time,
+        Some(custom_origin_time)
+    );
+    assert_eq!(
+        installed_app_role_1.dna.modifiers.quantum_time,
+        Some(custom_quantum_time)
+    );
+
+    assert_eq!(
+        installed_app_role_2.dna.modifiers.network_seed,
+        Some(network_seed_override.into())
+    );
+    assert_eq!(
+        installed_app_role_2.dna.modifiers.properties,
+        Some(manifest_properties.clone())
+    );
+    assert_eq!(
+        installed_app_role_2.dna.modifiers.origin_time,
+        Some(manifest_origin_time)
+    );
+    assert_eq!(
+        installed_app_role_2.dna.modifiers.quantum_time,
+        Some(manifest_quantum_time)
+    );
+
+    //- Test that modifier fields that are None in the modifiers map do not overwrite existing
```

Comment:
Consider splitting into a new test at this point

Reply:
I did not split because I thought that starting the conductor is costly and so running it all in one go runs faster. Is this a wrong assumption or should we just sacrifice this benefit for readability?

Reply:
As discussed in chat, yes please to splitting this up, for readability and for the sake of maintenance

Reply:
okay, have split the tests up now.